Add Semantics link in the SemanticsDebugger

mehere opened the pull request:
## Description

Fixed SemanticsDebugger that does not show Semantics link
Note that lots of diffs are due to dartfmt

## Related Issues

#44543

## Tests

I added the following tests:

Added tests to semantics_debugger_test.dart

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

mehere committed:
Add Semantics link in the SemanticsDebugger

goderbauer commented:
/cc @darrenaustin Is `SemanticsFlag.isLink` included in your Semantics Debugger update?

goderbauer commented:
@mehere As documented in our style guide, the flutter repository unfortunately doesn't use dartfmt. Can you please revert all the format changes introduced by it? Thanks.

Hixie committed:
Apply void_checks lint. (#48267)

engine-flutter-autoroll committed:
Roll engine 3d37d39d95ad..88935ca3c4fe (2 commits) (#48648)


https://github.com/flutter/engine/compare/3d37d39d95ad...88935ca3c4fe

git log 3d37d39d95ad..88935ca3c4fe --first-parent --oneline
2020-01-11 nurhan@google.com Support Clipboard.getData and Clipboard.setData for Flutter for web  (flutter/engine#15455)
2020-01-11 skia-flutter-autoroll@skia.org Roll src/third_party/dart 395daaa3ecfc..b3457c8cc2c1 (66 commits) (flutter/engine#15485)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

goderbauer committed:
Make ParentDataWidget usable with different ancestor RenderObjectWidget types (#48541)

zanderso committed:
[flutter_tool] Make BotDetector context free (#48605)

gspencergoog committed:
Allow requestFocus on an unattached FocusNode to create a deferred focus request (#48589)

This changes the behavior of requestFocus when it is called on a FocusNode that does not yet have a parent, so that it defers requesting focus until it receives a parent. Before this change, calling requestFocus before it had a parent was a no-op.

This allows scenarios where a widget is newly added and wishes to immediately request the focus. Previously, it was very hard to make that work because requesting focus before the widget's focus node had a parent was ignored, so the developer had to wait until two frames later to request focus (one for the widget's node to be added to the focus tree, and one to request the focus).

Now, in order to have a widget be focused when initially added, you just need to call requestFocus on its node when you create it, and as soon as it is added, it will automatically request focus.

This is different from the autofocus attribute on the Focus widget, because it unconditionally requests focus when added (autofocus will only request focus if nothing else in the scope has focus).

Piinks committed:
Fixing PageScrollPhysics to get along with NestedScrollView (#48457)

engine-flutter-autoroll committed:
Roll engine 88935ca3c4fe..73d2fb374ccb (3 commits) (#48733)


https://github.com/flutter/engine/compare/88935ca3c4fe...73d2fb374ccb

git log 88935ca3c4fe..73d2fb374ccb --first-parent --oneline
2020-01-11 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from 75iyW... to Xh8_9... (flutter/engine#15506)
2020-01-11 skia-flutter-autoroll@skia.org Roll src/third_party/dart b3457c8cc2c1..32ce9c7872e2 (7 commits) (flutter/engine#15504)
2020-01-11 skia-flutter-autoroll@skia.org Roll src/third_party/skia 3723fb7e85bb..0a1923e50752 (6 commits) (flutter/engine#15500)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

pennzht committed:
Add asserts requiring BoxConstraints' parameters to be non-null (#48295)

zanderso committed:
[flutter_tools] File system utilities (#48738)

zanderso committed:
[flutter_tools] File system utilities (#48738)

jakemac53 committed:
Serve packages uris in flutter_tools dev web server (#48743)

* support mapping /packages/<package>/<path> requests to package:<package>/<path> uris in the web device file server

zanderso committed:
Revert "[flutter_tools] File system utilities (#48738)" (#48755)

This reverts commit 034966062249c490e12ff252828e4ba4993bac2b.

zanderso committed:
Revert "[flutter_tools] File system utilities (#48738)" (#48755)

This reverts commit 034966062249c490e12ff252828e4ba4993bac2b.

mannprerak2 committed:
Passes scrollPhysics (#48342)

dnfield committed:
Fix tool test order (#48735)

Commit:
[flutter_tools][web] Add basic service worker generation support to web applications (#48344)

feinstein committed:
Fixing AutomaticKeepAlive Docs (#48575)

zanderso committed:
[flutter_tool] Roll the Fuchsia SDK forward (#48750)

engine-flutter-autoroll committed:
Roll engine 73d2fb374ccb..157772a31597 (4 commits) (#48770)


https://github.com/flutter/engine/compare/73d2fb374ccb...157772a31597

git log 73d2fb374ccb..157772a31597 --first-parent --oneline
2020-01-13 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from Xh8_9... to oxHfW... (flutter/engine#15547)
2020-01-13 skia-flutter-autoroll@skia.org Roll src/third_party/dart 32ce9c7872e2..28c335d5a201 (2 commits) (flutter/engine#15513)
2020-01-13 garyq@google.com Trivial typo fix to kick tests. (flutter/engine#15557)
2020-01-11 skia-flutter-autoroll@skia.org Roll src/third_party/skia 0a1923e50752..69ace2a0adbe (1 commits) (flutter/engine#15507)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

Commit:
[flutter_tools] Refactor Environment and FileStore to be context-free (#48759)

engine-flutter-autoroll committed:
Roll engine 157772a31597..0235a5084389 (3 commits) (#48777)


https://github.com/flutter/engine/compare/157772a31597...0235a5084389

git log 157772a31597..0235a5084389 --first-parent --oneline
2020-01-14 jason-simmons@users.noreply.github.com Change the value of EXTRA_INITIAL_ROUTE to match the value used by the old embedding and flutter_tools (flutter/engine#15569)
2020-01-13 jason-simmons@users.noreply.github.com [SkParagraph] Copy font features to the Skia layout engine text style (flutter/engine#15562)
2020-01-13 jason-simmons@users.noreply.github.com Add a type parameter to SavedState.Creator in the new embedding (flutter/engine#15563)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

xster committed:
Let the tool warn about all old API plugins (#48282)

mehere committed:
Add Semantics link in the SemanticsDebugger

mehere commented:
messed up with my branch. Created a new pull request instead: https://github.com/flutter/flutter/pull/48810
